src/elements/paginator/paginator.ts

```diff
@@ -0,0 +1,345 @@
+  public set pageSize(value: number) {
+    const previousPageSize = this._pageSize;
+    this._pageSize = Math.max(value, 0);
+    this.pageIndex = Math.floor((this.pageIndex * previousPageSize) / this._pageSize) || 0;
+  }
+  public get pageSize(): number {
+    return this._pageSize;
+  }
+  private _pageSize: number = 10;
+
+  /** Current page index. */
+  @property({ attribute: 'page-index', type: Number })
+  public set pageIndex(value: number) {
+    const previousPageIndex = this._pageIndex;
+    this._pageIndex = this._validatePageIndex(value);
+    if (previousPageIndex !== this._pageIndex) {
+      this._pageChanged.emit({ previousPageIndex, currentPageIndex: value });
+    }
+  }
+  public get pageIndex(): number {
+    return this._pageIndex;
+  }
+  private _pageIndex: number = 0;
+
+  /** The available `pageSize` choices. */
+  @property({ attribute: 'page-size-options', type: Array }) public pageSizeOptions?: number[];
+
+  /** Position of the prev/next buttons. */
+  @property({ attribute: 'pager-position' }) public pagerPosition: 'start' | 'end' = 'start';
+
+  /** Size variant, either m or s. */
+  @property({ reflect: true }) public size: 'm' | 's' = 'm';
+
+  private _pageChanged: EventEmitter<SbbPaginatorPageChanged> = new EventEmitter(
+    this,
+    SbbPaginatorElement.events.pageChanged,
+    { composed: true, bubbles: true },
+  );
+
+  private _abort = new SbbConnectedAbortController(this);
+  private _language = new SbbLanguageController(this);
+  private _markForFocus: number | null = null;
+
+  protected override firstUpdated(changedProperties: PropertyValues<this>): void {
+    super.firstUpdated(changedProperties);
+    const signal = this._abort.signal;
+    this.shadowRoot!.querySelector('.sbb-paginator__pages')!.addEventListener(
+      'keydown',
+      (e) => this._handleKeydown(e as KeyboardEvent),
+      { signal },
+    );
+  }
+
+  protected override updated(changedProperties: PropertyValues<this>): void {
+    super.updated(changedProperties);
+
+    /** Arrow navigation can force a rerender when ellipsis elements need to be displayed; the focus must stay on the correct element. */
+    if (this._markForFocus && sbbInputModalityDetector.mostRecentModality === 'keyboard') {
+      const focusElement = this._getVisiblePages().find(
+        (e) => this.pageIndex === +e.getAttribute('data-index')!,
+      );
+      if (focusElement) {
+        (focusElement as HTMLElement).focus();
+      }
+      // Reset mark for focus
+      this._markForFocus = null;
+    }
+  }
+
+  /** Returns the displayed page elements. */
+  private _getVisiblePages(): Element[] {
+    return Array.from(this.shadowRoot!.querySelectorAll('.sbb-paginator__page--number-item'));
+  }
+
+  /** Change page by setting the `pageIndex`. */
+  private _changePage(value: number): void {
+    this.pageIndex = value;
+  }
+
+  /** Evaluate `pageIndex` by excluding edge cases. */
+  private _validatePageIndex(pageIndex: number): number {
```

Comment:
First naming was 'calculate', i changed after https://github.com/sbb-design-systems/lyne-components/pull/2982#discussion_r1728673571 :D

Reply:
another proposal could be `checkPageIndexBounds`

Reply:
or clampPageIndex? or normalizePageIndex?
btw. should pageIndex > this._numberOfPages not result in this._numberOfPages ? and for the numberOfPages check shouldn't it be `numberOfPages - 1` ?

Reply:
Angular has `_coercePageIndexInRange` (the point about the `numberOfPages - 1` is right, i'm fixing it)
should we have a poll for naming? :D otherwise i propose the 'coerce' for consistency between libs

Reply:
oh I like the coerce and also the "inRange". now I'm happy (without a poll :P ):D